docs/functionalabstractions/concretetypes/index.md

```diff
@@ -3,4 +3,111 @@ id: index
+
+### Debug
+
+The `Debug` abstraction describes a type that can be rendered for debugging purposes. Its fundamental representation is as a function `A => Repr`, where `A` is a structured representation of the value that supports rendering in various formats.
+
+The `Debug` abstraction allows us to define how a data type should be rendered, for example rendering an `Array` to display its values rather than just its memory location. The more structured representation also allows rendering in different formats, for example including fully qualified names in the rendering of the data type so the rendering is itself valid Scala code.
+
+## Ways Of Combining Concrete Types
+
+The second set of abstractions describe different ways of combining two values of a given type. These can all be thought of as functions `(A, A) => A`.
+
+Combining is quite fundamental because it is how we build more complex structures from simpler ones.
+
+Generally unary operators are not very interesting because they don't add additional structure, they just modify existing structure.
+
+In contrast, given the ability to combine two values of a given type we can create a new value of that type with richer structure. And if we can do that once we can do it again and again after that, creating values with arbitrarily complex structure.
+
+The abstractions in ZIO Prelude for describing combining values are each based on fundamental algebraic properties that the combining operation can have.
+
+### Associative
+
+The most basic such abstraction is `Associative`, which describes a combining operation that is associative.
+
+```scala
+(a <> b) <> c === a <> (b <> c)
+```
+
+This essentially means that if we are combining three values then the order of operations doesn't matter. We can combine `a` and `b` and then combine the result with `c` or we can combine `b` and `c` and then combine `a` with the result.
+
+This is quite an important property for being able to build more complex structures from simpler ones. Otherwise the result of sequentially building more complex structures from simpler ones is not well defined because it depends on the order of operations and there are an exponentially increasing number of ways we could order these operations as we combine more and more values.
+
+There are also an extremely large number of data types that can be combined using associative operations.
+
+For example integers can be combined associatively using addition and multiplication, among other operations. String concatenation is also associative, as is concatenation of lists of any concrete type.
+
+For these reasons the `Associative` abstraction forms the root of the set of abstractions for describing ways to combine concrete types in ZIO Prelude.
+
+### Commutative
```

Comment:
I suppose that there's no interest/need of mentioning/warning that in classic math being commutative does not entail being associative as well?

Reply:
I think maybe it is more appropriate for the detailed section than this summary.